### PR TITLE
refactor: {lifecycle} instead of current deprecations

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

File: R/attributes.R [hunk 1]
```diff
@@ -1,3 +1,228 @@
...
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' `get.edge.attribute()` was renamed to `edge_attr()` to create a more
+#' consistent API.
+#' @inheritParams edge_attr
+#' @keywords internal
+#' @export
+get.edge.attribute <- function(graph , name , index = E(graph)) { # nocov start
+   lifecycle::deprecate_soft("1.6.0", "get.edge.attribute()", "edge_attr()")
```
Comment (krlmlr) [inline R/attributes.R:223]:
```suggestion
get.edge.attribute <- function(graph, name, index = E(graph)) { # nocov start
   lifecycle::deprecate_soft("2.0.0", "get.edge.attribute()", "edge_attr()")
```

File: R/adjacency.R [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+#'
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' `graph.adjacency()` was renamed to `graph_from_adjacency_matrix()` to create a more
+#' consistent API.
+#' @inheritParams graph_from_adjacency_matrix
+#' @keywords internal
+#' @export
+graph.adjacency <- function(adjmatrix , mode = c("directed","undirected","max","min","upper","lower","plus") , weighted = NULL , diag = TRUE , add.colnames = NULL , add.rownames = NA) { # nocov start
```
Comment (krlmlr) [inline R/adjacency.R:12]:
```suggestion
graph.adjacency <- function(adjmatrix, mode = c("directed", "undirected", "max", "min", "upper", "lower", "plus"), weighted = NULL, diag = TRUE, add.colnames = NULL, add.rownames = NA) { # nocov start
```

**Review by krlmlr — COMMENTED** [review]
Thanks! I like how we now can cherry-pick changes.

I saw issues with the spacing, and 1.6.0 is out already. Can you please regenerate?

**Comment by maelle** [comment]
(I'll open a new PR)

**Pull request closed without merging**